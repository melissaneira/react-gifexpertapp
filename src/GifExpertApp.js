import React, { useState } from 'react'
import CategoryAdd from './components/CategoryAdd'
import GifGrid from './components/GifGrid'

const GifExpertApp = ({defaultCategories =[]}) => {    
    const [categories, setCategories] =useState(defaultCategories);
     
    return (
        <>
            <h2>GifExpertApp</h2>
            <hr />
            <CategoryAdd setCategories={setCategories}></CategoryAdd>
            <ol>
                {categories.map(category =>{                    
                    return <GifGrid key={category} category={category}></GifGrid>
                })}
            </ol>            
        </>
      )
}

export default GifExpertApp;
