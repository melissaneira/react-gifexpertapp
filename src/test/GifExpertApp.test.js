import { shallow } from "enzyme"
import GifExpertApp from "../GifExpertApp"

describe('Probando componente GifExpertApp ', () => { 
    const wrapper = shallow(<GifExpertApp />);
    test('Debe renderizar el componente GifExpertApp', () => {         
        expect(wrapper).toMatchSnapshot();
     })

     test('Debe de mostrar una lista de categorias', () => {         
        const categories = ['one Punch', 'sailor moon', 'Goku'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toEqual(3);
        //console.log(wrapper);
     })
 })
 