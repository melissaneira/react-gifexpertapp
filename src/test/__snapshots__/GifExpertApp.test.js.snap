// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente GifExpertApp  Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <CategoryAdd
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="one Punch"
      key="one Punch"
    />
    <GifGrid
      category="sailor moon"
      key="sailor moon"
    />
    <GifGrid
      category="Goku"
      key="Goku"
    />
  </ol>
</Fragment>
`;

exports[`Probando componente GifExpertApp  Debe renderizar el componente GifExpertApp 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <CategoryAdd
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
